importer/likutim/create_unit.go

```diff
@@ -0,0 +1,237 @@
+package likutim
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/pkg/errors"
+	"github.com/spf13/viper"
+	"github.com/volatiletech/sqlboiler/boil"
+	"github.com/volatiletech/sqlboiler/queries/qm"
+	"gopkg.in/volatiletech/null.v6"
+
+	"github.com/Bnei-Baruch/mdb/common"
+	"github.com/Bnei-Baruch/mdb/models"
+	"github.com/Bnei-Baruch/mdb/utils"
+)
+
+type CreateUnits struct {
+	duplicates []*Double
+	mdb        *sql.DB
+}
+
+func (c *CreateUnits) Run() {
+	err := c.duplicatesFromJSON()
+	if err != nil {
+		log.Errorf("Error on read. Error: %s", err)
+		compare := new(Compare)
+		compare.Run()
+		c.duplicates = compare.result
+	}
+
+	c.openDB()
+	defer c.mdb.Close()
+
+	err = os.MkdirAll(viper.GetString("likutim.results-dir"), os.ModePerm)
+	if err != nil {
+		log.Errorf("Can't create directory: %s", err)
+	}
+	c.updateUnits()
+}
+
+func (c *CreateUnits) updateUnits() {
+	for _, d := range c.duplicates {
+		log.Debugf("\n\nStart create new unit type LIKUTIM, file: %s all files: %v", d.Save, d.Doubles)
+
+		tx, err := c.mdb.Begin()
+		if err != nil {
+			log.Error(err)
+			continue
+		}
+		err, cu := c.createUnitOnTransaction(tx, d.Save)
+		if err != nil {
+			log.Error(err)
+			tx.Rollback()
+			continue
+		}
+		err = tx.Commit()
+		// Derivation unit type LIKITIM to origins of doubles
+		for _, uid := range d.Doubles {
+			if uid == d.Save {
+				continue
+			}
+			tx, err := c.mdb.Begin()
+			if err != nil {
+				log.Error(err)
+				tx.Rollback()
+				continue
+			}
+			err = moveDataOnTransaction(tx, uid, cu)
+			err = tx.Commit()
+			if err != nil {
+				log.Errorf("problem commit transaction. Error: %s", err)
+			}
+			log.Debugf("End create new unit type LIKUTIM id: %d", cu.ID)
+		}
+	}
+}
+
+func (c *CreateUnits) createUnitOnTransaction(tx *sql.Tx, uid string) (error, *models.ContentUnit) {
```

Comment:
Error should be the last returned value. Not the first.

Reply:
This function doesn't manage transactions (begin, commit, rollback) so no need to mention that in the name.
IIRC sql.Tx implements boil.Executor so you could rename it as well.
grep for `exec boil.Executor` I'm using it in many places. 
See [this for example](https://github.com/Bnei-Baruch/mdb/blob/33511f2b4f511518dee1d77f3c35f35adaa040a6/api/handlers.go#L1196)